### PR TITLE
Zeppelin interpreter documentation and interpreter REST API documentation

### PR DESCRIPTION
**Pull request by eranwitkon**
Add documentation about the role of Interpreters, interpreters group and interpreters settings in Zeppelin and docs about interpreter REST API.

Please review verify that 
https://github.com/eranwitkon/incubator-zeppelin/blob/gh-pages/docs/manual/interpreters.md
is correct...

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ tagline: Less Development, More analysis!
 ### Multiple language backend
 
 Zeppelin interpreter concept allows any language/data-processing-backend to be plugged into Zeppelin.
-Current languages included in the Zeppelin interpreter are: Scala(with Apache Spark), Python(with Apache Spark), SparkSQL, Hive, Markdown and Shell.
+Currently Zeppelin support many interpreters such as Scala(with Apache Spark), Python(with Apache Spark), SparkSQL, Hive, Markdown and Shell.
```
Comment (felixcheung) [inline index.md]: minor: "support" -> "supports"

File: docs/manual/interpreters.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Interpreters in zeppelin
+
+This section explain the role of Interpreters, interpreters group and interpreters settings in Zeppelin.
+Zeppelin interpreter concept allows any language/data-processing-backend to be plugged into Zeppelin.
+Currently Zeppelin support many interpreters such as Scala(with Apache Spark), Python(with Apache Spark), SparkSQL, Hive, Markdown and Shell.
+
+### What is zeppelin interpreter?
+
+Zeppelin Interpreter is the plug-in which enable zeppelin user to use a specific language backend. For example to use scala code in Zeppelin, you need scala interpreter.
```
Comment (felixcheung) [inline docs/manual/interpreters.md]: Is this confusing? there isn't a "Scala" interpreter - there is only a Spark interpreter on the list

Comment (eranwitkon) [inline docs/manual/interpreters.md]: Both fixed. Thanks for the review!
Eran

**Comment by Leemoonsoo** [comment]
Looks great. +1

File: docs/manual/interpreters.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+<img src="../../assets/themes/zeppelin/img/screenshots/interpreter_setting.png">
+### What is zeppelin interpreter group?
+
+Every Interpreter belongs to an InterpreterGroup. InterpreterGroup is a unit of start/stop interpreter.
+By default, every interpreter belong to a single group but the group might contain more interpreters. For example, spark interpreter group include spark support, pySpark, 
+SparkSQL and the dependency loader.
+
+Technically, Zeppelin interpreters from the same group are running in the same JVM.
+
+**TODO: Validate this with the community ...**
```
Comment (bzz) [inline docs/manual/interpreters.md]: Do we want to keep if for now, or you need some kind of input to resolve it?

File: docs/rest-api/rest-interpreter.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+---
+{% include JB/setup %}
+
+## Zeppelin REST API
+ Zeppelin provides several REST API's for interaction and remote activation of zeppelin functionality.
+ 
+ All REST API are available starting with the following endpoint ```http://[zeppelin-server]:[zeppelin-port]/api```
+ 
+ Note that zeppein REST API receive or return JSON objects, it it recomended you install some JSON view such as 
+ [JSONView](https://chrome.google.com/webstore/detail/jsonview/chklaanhfefbnpoihckbnefhakgolnmc?utm_source=chrome-app-launcher-info-dialog)
```
Comment (bzz) [inline docs/rest-api/rest-interpreter.md]: If we want to keep a link to the external resource, may be we should remove the Google Analytics tracking part at least `?utm_source=chrome-app-launcher-info-dialog`

**Comment by bzz** [comment]
Thank you very much for the contribution! 
For the REST API doc - I wish we could use swagger...

File: docs/manual/interpreters.md (hunk 3 above)
Comment (eranwitkon) [inline docs/manual/interpreters.md]: I wasn't sure that the description of what is a registered interpreter,
setting and group mean.
If it is correct I will remove the TODO and the url parameter as commented.
BTW, Internally, I think the doing a create setting which is actualky
calling add group is a bed naming and hard to follow.
Has no effect on documentation for now...

בתאריך יום ו׳, 28 באוג׳ 2015, 09:05 מאת Alexander <notifications@github.com

**Comment by Leemoonsoo** [comment]
@bzz recent discussion related to swagger https://github.com/apache/incubator-zeppelin/pull/226

**Comment by eranwitkon** [comment]
Docs fixed according to comments. 
Ready for merge.

**Comment by eranwitkon** [comment]
Docs fixed abd ready for merge.

בתאריך יום ו׳, 28 באוג׳ 2015, 09:21 מאת Lee moon soo <
notifications@github.com>:

**Comment by Leemoonsoo** [comment]
LGTM

**Comment by bzz** [comment]
@Leemoonsoo thank you for pointing out the history!

Looks great to me.
